Prometheus: More consistently allows for multi-line queries in editor

kaydelaney opened the pull request:
**What this PR does / why we need it**:
Allows a user to hit shift+enter to create a new line in the query field, even
when the autocomplete suggestions are displayed.

**Which issue(s) this PR fixes**:
Closes #18341

torkelo commented on public/app/features/explore/QueryField.tsx:
```diff
@@ -313,32 +313,35 @@ export class QueryField extends React.PureComponent<QueryFieldProps, QueryFieldS
-      if (!suggestions || suggestions.length === 0) {
-        return undefined;
-      }
+    if (!event.shiftKey) {
```
personally I usually try to limit code nesting, as it usually leads to more readable code. So in this case I would just check if (event.shiftKey) and return (at the top of function), instead of indenting everything in this negative if case.

davkal replied:
Yeah, I'd exit early here too.

kaydelaney replied:
definitely agree, i usually try to limit nesting as well but seem to have missed it this time, oops! will push another commit.

davkal reviewed (commented):
Works well! But needs a rebase, sorry for the delay!

kaydelaney committed:
Explore/Prometheus: More consistently allows for multi-line queries
Allows a user to hit shift+enter to create a new line in the query field, even
when the autocomplete suggestions are displayed.
Also fixes an issue where a new line was inserted when selecting a suggestion
Closes #18341

davkal reviewed (changes requested):
Running queries with Return should work, in this branch it currently does a new line (same as Shift+Return).

kaydelaney committed:
Fixes behavior where query wasn't running on pressing Enter
Also adds test to verify this behavior

davkal commented:
Looks good to merge now.
